contracts/PaymentReceiver.sol

```diff
@@ -0,0 +1,63 @@
+pragma solidity ^0.8.11;
+
+import "hardhat/console.sol";
+import "./IPaymentReceiver.sol";
+
+contract ClientDatabase {
+    uint private nonce;
+    mapping (uint256 => address) private clientIds;
+    uint256[] private idList;
+    // An address cannot have more than 128 associated clients.
+    uint32 private maxClientIds = 128;
+
+    constructor() {}
+
+    function addClient() external returns (uint256) {
+        uint256 _clientId = uint256(keccak256(abi.encode(block.number, msg.data, nonce++)));
```

Comment:
Any reason you would use a random clientID as opposed to an incremental? This costs extra gas. I would suggest the usage of the Counter class